types/inquirer/index.d.ts

```diff
@@ -70,90 +72,158 @@ declare namespace inquirer {
-         * (to save in the answers hash). Values can also be a Separator.
+         * Change the default _suffix_ message.
          */
-        choices?:
-        | ReadonlyArray<ChoiceType>
-        | ((answers: T) => ReadonlyArray<ChoiceType>)
-        | ((answers: T) => Promise<ReadonlyArray<ChoiceType>>);
+        suffix?: string;
         /**
-         * Receive the user input and should return true if the value is valid, and an error message (String)
-         * otherwise. If false is returned, a default error message is provided.
+         * Receive the current user answers hash and should return `true` or `false` depending
+         * on whether or not this question should be asked. The value can also be a simple boolean.
          */
-        validate?(input: any, answers?: T): boolean | string | Promise<boolean | string>;
+        when?: boolean | ((answers: A) => boolean | Promise<boolean>);
+    }
+
+    interface QuestionOptions<A> {
+        /**
+         * Choices array or a function returning a choices array. If defined as a function,
+         * the first parameter will be the current inquirer session answers. Array values can
+         * be simple `numbers`, `strings`, or `objects` containing a `name` (to display in list),
+         * a `value` (to save in the answers hash) and a `short` (to display after selection)
+         * properties. The choices array can also contain
+         * [a Separator](https://github.com/SBoudrias/Inquirer.js#separator).
+         */
+        choices?:
+        | ReadonlyArray<ChoiceType<A>>
+        | ((answers: A) => ReadonlyArray<ChoiceType<A>>)
+        | ((answers: A) => Promise<ReadonlyArray<ChoiceType<A>>>);
         /**
          * Receive the user input and return the filtered value to be used inside the program.
-         * The value returned will be added to the Answers hash.
+         * The value returned will be added to the _Answers_ hash.
          */
         filter?(input: string): any;
         /**
-         * Receive the user input and return the transformed value to be displayed to the user. The
-         * transformation only impacts what is shown while editing. It does not impact the answers
-         * hash.
+         * Receive the user input, answers hash and option flags, and return a transformed value
+         * to display to the user. The transformation only impacts what is shown while editing.
+         * It does not modify the answers hash.
          */
-        transformer?(input: string): string;
+        transformer?(input: string, answers: A, flags: any): string;
+        /**
+         * Change the number of lines that will be rendered when using `list`, `rawList`,
+         * `expand` or `checkbox`.
+         */
+        pageSize?: number;
+    }
+
+    type Question<A extends Answers = Answers, T extends Record<string, any> = {}> = (
+        ListQuestion<A> & T |
```

Comment:
Oooh I see what you planned to do
But I think another good option would be to drop the additional generic parameter and use
```ts
type GeneratorQuestion = Question<A> & {
    /** comment stuff */
    store?: boolean
}
```
in the yeoman-generator instead.
What do you think about this idea?

Reply:
Yeah, I like that much better. See dc59344!